deps/chakrashim/src/jsrtcontextshim.cc

```diff
@@ -447,6 +451,46 @@ bool ContextShim::ExecuteChakraShimJS() {
                         &result) == JsNoError;
 }
 
+bool ContextShim::ExecuteChakraDebugShimJS(JsValueRef * chakraDebugObject) {
```

Comment:
I though of that but that will make things a bit complicated as we need to allocate buffer using new and then in all failure cases make sure it is deleted, Also there not not much command code. We call different APIs in both functions. For now I think we used keep it as it is and if we add more shim then we should definitely refactor this.

---
In reply to: [82066039](https://github.com/nodejs/node-chakracore/pull/126#discussion_r82066039) [](ancestors = 82066039)